src/lazy/text/raw/sequence.rs

```diff
@@ -163,106 +139,109 @@ impl<'data> Iterator for RawTextListIterator_1_0<'data> {
-                ))))
-            }
-            Ok((_remaining, None)) => None,
+        // Copy the original input so we can include any matched annotations.
+        let input = self.input;
+        let result = whitespace_and_then(alt((
+            // We only peek at the end so future calls to `next()` will continue to yield `None`.
+            peek(")").value(None),
+            // An annotated value or (in Ion 1.1) an e-expression
+            E::value_expr_matcher().map(Some),
+            // A potentially annotated operator literal
+            (
+                opt(TextBuffer::match_annotations),
+                whitespace_and_then(TextBuffer::match_operator),
+            )
+                .map(|(maybe_annotations, value)| input.apply_annotations(maybe_annotations, value))
+                .map(RawValueExpr::ValueLiteral)
+                .map(Some),
+        )))
+        .parse_next(&mut self.input);
+
+        match result {
+            Ok(Some(value_expr)) => Some(Ok(value_expr)),
+            Ok(None) => None,
             Err(e) => {
                 self.has_returned_error = true;
-                e.with_context("reading the next sexp value", self.input)
+                e.with_context("reading the next s-expression value", self.input)
                     .transpose()
             }
         }
     }
+
+    // fn next(&mut self) -> Option<Self::Item> {
+    //     if self.has_returned_error {
+    //         return None;
+    //     }
+    //     match self.input.match_sexp_item() {
+    //         Ok(Some(value)) => Some(Ok(RawValueExpr::ValueLiteral(LazyRawTextValue::<E>::from(
+    //             value,
+    //         )))),
+    //         Ok(None) => None,
+    //         Err(e) => {
+    //             self.has_returned_error = true;
+    //             e.with_context("reading the next sexp value", self.input)
+    //                 .transpose()
+    //         }
+    //     }
+    // }
 }
 
-impl<'data> LazyContainerPrivate<'data, TextEncoding_1_0> for LazyRawTextSExp_1_0<'data> {
-    fn from_value(value: LazyRawTextValue_1_0<'data>) -> Self {
-        LazyRawTextSExp_1_0 { value }
+impl<'data, E: TextEncoding<'data>> LazyContainerPrivate<'data, E> for RawTextSExp<'data, E> {
+    fn from_value(value: LazyRawTextValue<'data, E>) -> Self {
+        RawTextSExp { value }
     }
 }
 
-impl<'data> LazyRawContainer<'data, TextEncoding_1_0> for LazyRawTextSExp_1_0<'data> {
-    fn as_value(&self) -> <TextEncoding_1_0 as Decoder>::Value<'data> {
+impl<'data, E: TextEncoding<'data>> LazyRawContainer<'data, E> for RawTextSExp<'data, E> {
+    fn as_value(&self) -> <E as Decoder>::Value<'data> {
         self.value
     }
 }
 
-impl<'data> LazyRawSequence<'data, TextEncoding_1_0> for LazyRawTextSExp_1_0<'data> {
-    type Iterator = RawTextSExpIterator_1_0<'data>;
+impl<'data, E: TextEncoding<'data>> LazyRawSequence<'data, E> for RawTextSExp<'data, E> {
+    type Iterator = RawTextSequenceCacheIterator<'data, E>;
```

Comment:
🪧 All of the text containers now cache their child expressions and iterate over the cache as needed.